### PR TITLE
Pr 2020.01

### PR DESCRIPTION
**Pull request by montjoie**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -309,8 +309,10 @@ boards:
     kvm: (For qemu only) Does the qemu could use KVM (default: no)
     uboot_ipaddr:	(optional) a static IP to set in uboot
     uboot_macaddr:	(Optional) the MAC address to set in uboot
-    custom_option:	(optional) All following strings will be directly append to devicefile
+    custom_option:	(optional) All following strings will be directly append to devicefile included in {% opt %}
     - "set x=1"
+    raw_custom_option:	(optional) All following strings will be directly append to devicefile
+    - "{% set x=1 %}"
```
Comment (khilman) [inline README.md]: YAML lets you pass raw data already with `|-`.  e.g.
```
    custom_option: |-
       set bootloader_prompt = "=>"
       set boot_character_delay = 10

```

Can't you use that?

File: lavalab-gen.py [hunk 2]
```diff
@@ -155,6 +156,12 @@ def main():
         groupdir = "%s/groups" % workerdir
         os.mkdir(groupdir)
         worker = master
+        if "version" in worker:
+            dockerfile = open("%s/Dockerfile" % workerdir, "r+")
+            dockerfilec = re.sub('(^FROM.*:).*', '\g<1>%s' % worker["version"], dockerfile.read())
+            dockerfile.seek(0)
```
Comment (khilman) [inline lavalab-gen.py:162]: This smells funny.  What if the new version string has more characters than the previous version string.   Aren't you going to potentially overwrite stuff in the updated Dockerfile?

Comment (khilman) [reply lavalab-gen.py:162]: I think the changelog needs more description on what exactly is being replaced in the Dockerfile

Comment (montjoie) [reply lavalab-gen.py:162]: I have added an example.
I have tested, and there no risk of overwrite. (the regex seems to stop at newline)

**Review by khilman — APPROVED** [review]

**Pull request merged**